pjsip/src/pjsip-ua/sip_reg.c

```diff
@@ -1359,6 +1359,14 @@ static void regc_tsx_callback(void *token, pjsip_event *event)
 						       PJSIP_REGC_MAX_CONTACT,
 						       contact);
 
+	    if (expiration == 0 && regc->current_op != REGC_UNREGISTERING) {
+		if (regc->expires_hdr && regc->expires_hdr->ivalue) {
+		    expiration = regc->expires_hdr->ivalue;
+		} else {
+		    expiration = PJSIP_REGISTER_EXP_REFRESH;
+		}
+	    }
```

Comment:
As this code handles a corner case:
- print some log?
- put a comment too, perhaps simply a reference to this PR

And as it introduces behavioral change/backward incompatibility:
  - consider a macro setting to disable this perhaps
  - please don't forget to mention it in the milestone after merged

Reply:
So suppose the registrar responses with:
case 1:
```
Contact: <sip:url1>;expires=0
```
case 2:
```
Contact: <sip:url1>;expires=0
Contact: <sip:url1>;expires=180
```

What expiration do we actually want to set, which will determine when we will try to reregister?
If it's too long (such as 3600), in case 2, it will have already expired, won't It?

In my opinion, it should be:
if server specifies expiration, min(PJSIP_REGISTER_EXP_REFRESH, the lowest non-zero expiration), else min (PJSIP_REGISTER_EXP_REFRESH, regc->expires_hdr->ivalue)

Reply:
Basically I am okay with any value, as this is for handling a corner case only, not really a general default value. So I just tried to point out there is already 'default value' (for cases that there is no expire value), not sure if we should have another and configurable (while the others are not). Also, 3600 seems to be the implicitly-recommended default value for non-determinable value (e.g: [here](https://datatracker.ietf.org/doc/html/rfc3261#section-10.2)). So IMO if we need to have a default value, I guess 3600 should be reasonable (btw, 30 is too small too, isn't it?).

For case 2 above, perhaps ideally we can use the 180s, but the modification needs to be done inside `calculate_response_expiration()`, e.g: skip the expire=0 in the iteration. But again I am okay too with the current patch :)

The sip_reg.c seems to have an 'algo' to decide the expire value if there is none, e.g: 
https://github.com/pjsip/pjproject/blob/ccc0644df69b7f9cf00de22cb1e4dec25d01413c/pjsip/src/pjsip-ua/sip_reg.c#L1025